pilot/pkg/proxy/envoy/v2/monitoring.go

```diff
@@ -0,0 +1,184 @@
+	ldsReject       = stats.Int64("pilot_xds_lds_reject", "Pilot rejected LDS.", stats.UnitDimensionless)
+	rdsReject       = stats.Int64("pilot_xds_rds_reject", "Pilot rejected RDS.", stats.UnitDimensionless)
+	rdsExpiredNonce = stats.Int64("pilot_rds_expired_nonce", "Total number of RDS messages with an expired nonce.", stats.UnitDimensionless)
+	totalXDSRejects = stats.Int64("pilot_total_xds_rejects", "Total number of XDS responses from pilot rejected by proxy.", stats.UnitDimensionless)
+	monServices     = stats.Int64("pilot_services", "Total services known to pilot.", stats.UnitDimensionless)
+
+	// TODO: Update all the resource stats in separate routine
+	// virtual services, destination rules, gateways, etc.
+	monVServices             = stats.Int64("pilot_virt_services", "Total virtual services known to pilot.", stats.UnitDimensionless)
+	xdsClients               = stats.Int64("pilot_xds", "Number of endpoints connected to this pilot using XDS.", stats.UnitDimensionless)
+	xdsResponseWriteTimeouts = stats.Int64("pilot_xds_write_timeout", "Pilot XDS response write timeouts.", stats.UnitDimensionless)
+	pushTimeouts             = stats.Int64("pilot_xds_push_timeout", "Pilot push timeout, will retry.", stats.UnitDimensionless)
+	pushTimeoutFailures      = stats.Int64("pilot_xds_push_timeout_failures", "Pilot push timeout failures after repeated attempts.", stats.UnitDimensionless)
+
+	// Covers xds_builderr and xds_senderr for xds in {lds, rds, cds, eds}.
+	pushes = stats.Int64("pilot_xds_pushes", "Pilot build and send errors for lds, rds, cds and eds.", stats.UnitDimensionless)
+	// values for pushes
+	cdsPushCtx, cdsSendErrPushCtx, cdsBuildErrPushCtx context.Context
+	edsPushCtx, edsSendErrPushCtx                     context.Context
+	ldsPushCtx, ldsSendErrPushCtx, ldsBuildErrPushCtx context.Context
+	rdsPushCtx, rdsSendErrPushCtx, rdsBuildErrPushCtx context.Context
+
+	pushErrors = stats.Int64("pilot_xds_push_errors", "Number of errors (timeouts) pushing to sidecars.", stats.UnitDimensionless)
+	// values for pushErrors
+	unrecoverableCtx, retryCtx context.Context
+
+	// only supported dimension is millis, unfortunately. default to unitdimensionless.
+	proxiesConvergeDelay     = stats.Float64("pilot_proxy_convergence_time", "Delay between config change and all proxies converging.", stats.UnitDimensionless)
+	proxiesConvergenceBounds = view.Distribution(1, 3, 5, 10, 20, 30, 50, 100)
+	pushContextErrors        = stats.Int64("pilot_xds_push_context_errors", "Number of errors (timeouts) initiating push context.", stats.UnitDimensionless)
+	totalXDSInternalErrors   = stats.Int64("pilot_total_xds_internal_errors", "Total number of internal XDS errors in pilot.", stats.UnitDimensionless)
+
+	inboundUpdates = stats.Int64("pilot_inbound_updates", "Total number of updates received by pilot.", stats.UnitDimensionless)
```

Comment:
This is dimensioned by type, why is it declared as dimensionless ?

Reply:
here, dimensions is a misnomer. this option is about the units of the value itself (bytes, seconds, etc.). OC only exposes the following:

```
    UnitDimensionless = "1"
    UnitBytes         = "By"
    UnitMilliseconds  = "ms"
```

A count of inbound updates is neither in a unit of bytes or milliseconds, making it "dimensionless" in OC lingo.